solver/src/settlement.rs

```diff
@@ -1,27 +1,36 @@
-use model::{Order, TokenPair, UserOrder};
-use std::{
-    collections::HashMap,
-    time::{Duration, Instant},
-};
+use model::UserOrder;
+use primitive_types::{H160, U256};
+use std::collections::HashMap;
 
-pub struct Settlement {
-    pub token_pair: TokenPair,
-    pub orders: Vec<UserOrder>,
-    // TODO: sell amounts, AMM trades
+#[derive(Debug)]
+pub struct Trade {
+    pub order: UserOrder,
+    pub executed_amount: U256,
+}
+
+#[derive(Debug)]
+pub enum Interaction {
```

Comment:
Maybe not needed immediately, but I think in the long run it might be better to model Interactions as a trait object and have different implementations of e.g. Uniswap, Curve, etc that can just return the calldata needed for "executing" the interaction.

With an enum I'm afraid this will become a large centralised data-structure.

Reply:
Let's do that when we see that it would be beneficial but I think you are probably right.